src/app/athkar-focus.tsx

```diff
@@ -0,0 +1,391 @@
+  const [showInstructions, setShowInstructions] = useState(true);
+  const [showSwipeIndicator, setShowSwipeIndicator] = useState(false);
+  const swipeIndicatorOpacity = useSharedValue(0);
+  const swipeIndicatorTranslateX = useSharedValue(0);
+  const [startFromIndex, setStartFromIndex] = useState(0);
+  const [currentAthkarIndex, setCurrentAthkarIndex] = useState(0);
+
+  // Animated values for circle progress
+  const animatedProgress = useSharedValue(0);
+  const circleScale = useSharedValue(1);
+  const countOpacity = useSharedValue(1);
+
+  const filteredAthkar = athkarList.filter(
+    (a) => a.type === currentType || a.type === ATHKAR_TYPE.ALL
+  );
+
+  useEffect(() => {
+    const index = athkarIndexToStartFrom(currentProgress, filteredAthkar, currentType);
+    setCurrentAthkarIndex(index);
+    setStartFromIndex(index);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [currentProgress, currentType]);
+
+  // Enable focus mode on mount
+  useEffect(() => {
+    toggleFocusMode();
+    return () => {
+      toggleFocusMode();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  // Hide instructions after 3 seconds
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setShowInstructions(false);
+    }, 3000);
+    return () => clearTimeout(timer);
+  }, []);
+
+  const currentAthkar = filteredAthkar[startFromIndex];
+  const progressItem = currentProgress.find(
+    (p) => p.athkarId === `${currentAthkar?.id}-${currentType}`
+  );
+  const currentCount = progressItem?.currentCount || 0;
+  const totalCount = currentAthkar?.count || 1;
+  const progressPercentage = (currentCount / totalCount) * 100;
+  const isCompleted = progressItem?.completed || false;
+
+  // Animate progress when count changes
+  useEffect(() => {
+    animatedProgress.value = withSpring(progressPercentage / 100, {
+      damping: 15,
+      stiffness: 150,
+      mass: 1,
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [progressPercentage]);
+
+  // Circle constants
+  const CIRCLE_RADIUS = 120;
+  const CIRCLE_CIRCUMFERENCE = 2 * Math.PI * CIRCLE_RADIUS;
+
+  // Animated props for the progress circle
+  const animatedCircleProps = useAnimatedProps(() => {
+    const strokeDashoffset = CIRCLE_CIRCUMFERENCE * (1 - animatedProgress.value);
+
+    return {
+      strokeDashoffset,
+      stroke: interpolateColor(
+        animatedProgress.value,
+        [0, 1],
+        ["#1E40AF", "#10b981"] // Blue to green
+      ),
+    };
+  });
+
+  // Animated style for circle scale effect
+  const circleAnimatedStyle = useAnimatedStyle(() => {
+    return {
+      transform: [{ scale: circleScale.value }],
+    };
+  });
+
+  // Animated style for count text
+  const countAnimatedStyle = useAnimatedStyle(() => {
+    return {
+      opacity: countOpacity.value,
+      transform: [{ scale: countOpacity.value }],
+    };
+  });
+
+  // Handle tap to increment
+  const handleTap = () => {
```

Comment:
So currently, when tapping, the change is abrupt, let's maybe have a swipe animation just to show that it has changed.
This way, we can even keep the progress indicator, and make it swipe along side the main content.

- Now when done:
    - progress reset quickly
    - text just change
- Suggested:
    - Swipe the whole content:
        - progress circle also swipped (as if we have 24 circles, but we can just simulate it)
        - content is swiped

With this also, if we implement swipe left and right for going next/perv, it will have smooth transition.